metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -2028,8 +2028,11 @@ public CompletableFuture<FinalizedControllerFeatures> finalizedFeatures(
         if (lastCommittedOffset == -1) {
             return CompletableFuture.completedFuture(new FinalizedControllerFeatures(Collections.emptyMap(), -1));
         }
+        // It's possible for a standby controller to receiving ApiVersionRequest and we do not have any timeline snapshot
+        // in a standby controller, in this case we use Long.MAX_VALUE.
+        long epoch = isActive() ? lastCommittedOffset : Long.MAX_VALUE;
```

Comment:
I wonder, would using `SnapshotRegistry.LATEST_EPOCH` make more sense semantically? We're also specifically checking for that value in `TimelineObject`.

Reply:
good suggestion.